crates/ntx-builder/src/state/account.rs

```diff
@@ -94,24 +95,22 @@ pub struct AccountState {
 }
 
 impl AccountState {
-    /// Creates a new account state using the given value as the committed state.
-    pub fn from_committed_account(account: Account) -> Self {
-        Self {
+    pub fn new(account: Account, notes: Vec<NetworkNote>) -> Self {
```

Comment:
Could we add a brief comment here explaining what this constructor expects and what it does? For example, my understanding is that it expects notes targeted to this account, but it ignores non-single target notes (should we panic on this?), and also does not check if the single-target notes are actually targeted to this account.